internal/terraform/terraform.go

```diff
@@ -0,0 +1,194 @@
+package terraform
+
+import (
+	"fmt"
+	"io/fs"
+	"log"
+	"os"
+	"path"
+	"path/filepath"
+	"strings"
+
+	"github.com/coopnorge/mage/internal/devtool"
+)
+
+
+// IsTerraformProject returns true if a directory contains a go module.
+func IsTerraformProject(p string, d fs.DirEntry) bool {
+	if !d.IsDir() {
+		return false
+	}
+	if _, err := os.Stat(path.Join(p, ".terraform.lock.hcl")); os.IsNotExist(err) {
+		return false
+	}
+	return true
+}
+
+// FindTerraformProjects will search through the base directory to find the all the
+// Go modules.
+func FindTerraformProjects(base string) ([]string, error) {
+	directories := []string{}
+
+	err := filepath.WalkDir(base, func(workDir string, d fs.DirEntry, err error) error {
+		if err != nil {
+			return err
+		}
+		if isDotDirectory(workDir, d) {
+			return nil
```

Comment:
If we find a dot-directory, we should probably not go into the dot-directory to find any other terraform-projects in there. 
```suggestion
			return filepath.SkipDir
```

Reply:
ok, good point, i did not know about SkipDir